### PR TITLE
Indexing PR prep

### PR DESCRIPTION
**Pull request by emetsger**
PR to get Travis' feedback on changes.  :)

### CODE REVIEW

**Comment by emetsger** [comment]
@karenhanson all decisions made in `DiscosIndexer` except for `DELETION` event types were made prior to lineage being available, so they definitely deserve to be revisited, and can be re-cast in terms of operations on the lineage.

Because there is no concept of a Solr transaction at the moment, it is possible to add a Solr document representing an `ACTIVE` DiSCO, and have the system crash before the previously `ACTIVE` DiSCO is inactivated.  I think we would need some kind of transaction/two-phase commit implementation to insure this doesn't happen.  Since we don't have that, the post processing doesn't necessarily make assumptions about how many ACTIVE documents to expect when deactivating or tombstoning.

If/when these operations are recast in terms of lineage (e.g. "tombstone this lineage"), probably not much will change, but at least the mental model ("operations on lineages") will be a little cleaner.

File: indexing-solr/src/main/java/info/rmapproject/indexing/solr/repository/DiscosIndexer.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                                && candidate.getDiscoStatus().equals(ACTIVE.name()));
+                break;
+
+            case DELETION:
+                discoOperations.deleteDocumentsForLineage(doc.getEventLineageProgenitorUri());
+                break;
+
+            case TOMBSTONE:
+                // FIXME: TOMBSTONE all (ACTIVE, INACTIVE) documents in the newly indexed document's lineage
+                discoOperations.updateStatus(doc.getDiscoUri(), RMapStatus.TOMBSTONED, null);
```
[the remark this answers is not in the document]
Comment (emetsger) [reply indexing-solr/src/main/java/info/rmapproject/indexing/solr/repository/DiscosIndexer.java]: Ok, so just making sure I understand:
1. Create Disco `A` (implies a lineage with progenitor `A` is created), with the "bad" data
2. Create Disco version `A'` with "clean" data, a `DERIVATION` of `A`, and implies a lineage with progenitor `A'` is created
3. Then, `TOMBSTONE` Disco `A`, which implies that it (or any previous versions, if they existed), should not be viewable.  

Is my interpretation above correct? 

If so, what if step 3 read:
3. Then, `TOMBSTONE` Disco `A`, which implies that its lineage should not be viewable.

[the remark this answers is not in the document]
Comment (karenhanson) [reply indexing-solr/src/main/java/info/rmapproject/indexing/solr/repository/DiscosIndexer.java]: Hopefully this clarifies things. These would all be on the same lineage progenitor. A DERIVATION is a whole new lineage, lineages do not affect each other.
```
//create discoV1
discoV1 (active)
//update discoV1
discoV1 (inactive)
discoV2 (active)
//update discoV2
discoV1 (inactive)
discoV2 (inactive)
discoV3 (active)
//update discoV2 
Cant update, the most recent version is discoV3
//tombstone discoV2
discoV1 (inactive)
discoV2 (tombstoned - but content in triplestore)
discoV3 (active)
//hard delete discoV3
discoV1 (inactive)
discoV2 (tombstoned - but content in triplestore)
discoV3 (deleted - content not in triplestore)
```

**Comment by karenhanson** [comment]
After a conversation, the reason for going through the lineage and setting ACTIVE->INACTIVE on an update makes good sense.  I've logged issue #176 for work on the other comment.

**Pull request merged**